### PR TITLE
FIX : Fix bug in mne.Epochs when passing events as list with event_id=None

### LINKED ISSUE #4822: BUG: default epochs events does not work with list
works:

```python
import numpy as np
epo = mne.Epochs(mne.io.RawArray(np.random.randn(10, 1000),
                                 mne.create_info(10, 1000.)),
                 events=np.array([[100, 0, 1], [200, 0, 1], [300, 0, 1]]))
```

does not work:
```python
import numpy as np
epo = mne.Epochs(mne.io.RawArray(np.random.randn(10, 1000),
                                 mne.create_info(10, 1000.)),
                 events=[[100, 0, 1], [200, 0, 1], [300, 0, 1]])
```

cc @larsoner

### PR DESCRIPTION
**Pull request by agramfort**
closes #4822

### CODE REVIEW

File: mne/tests/test_epochs.py [hunk 1]
```diff
@@ -2346,4 +2346,12 @@ def assert_metadata_equal(got, exp):
         assert check.all().all()
 
 
+def test_events_list():
+    """Tests what events can be a list."""
```
Comment (jona-sassenhagen) [inline mne/tests/test_epochs.py]: what -> that

File: mne/tests/test_epochs.py [hunk 2]
```diff
@@ -2346,4 +2346,12 @@ def assert_metadata_equal(got, exp):
...
+    events = [[100, 0, 1], [200, 0, 1], [300, 0, 1]]
+    epo = mne.Epochs(mne.io.RawArray(np.random.randn(10, 1000),
```
Comment (jona-sassenhagen) [inline mne/tests/test_epochs.py]: If I'm counting right, you can call it `epochs` ...

**Comment by agramfort** [comment]
good to go I think

**Comment by jona-sassenhagen** [comment]
I agree. What's with appveyor?

**Comment by agramfort** [comment]
appveyor somehow stalled. Travis is green so... I'll merge. Thanks @jona-sassenhagen for the review.

**Pull request merged**